src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
maybe i miss something, but all the methods except sendRequest seem like a repetition of the request factory to me. the client would be much simpler if it only had that one method and would not need to bother to be a factory as well.

especially with the topic of the body value objects - sounds like something for the request factory. the RequestInterface already handles all the stream and whatnot stuff. if the request factory is more useful with body value objects that is fine, but the client should not care imho.